addons/base_import_module/models/ir_module.py

```diff
@@ -399,18 +402,40 @@ def button_immediate_install_app(self):
 
     @api.model
     def _get_missing_dependencies(self, zip_data):
-        modules = self._get_missing_dependencies_modules(zip_data)
+        if release.version[-1] != 'e':  # Need to import release for that, not version from release
+            return _(
+                "The installation of the data module will fail as you need the Enterprise version "
+                "to install an industry module. Please visit https://www.odoo.com/pricing-plan for "
+                "more information.\n"
+            ), False
+        modules, unavailable_modules = self._get_missing_dependencies_modules(zip_data)
         description = ''
-        if modules:
-            description = _('The following modules will be also installed:\n')
+        if unavailable_modules:
+            description = _(
+                "The installation of the data module will fail as the following dependencies can't "
+                "be found in the addons-path:\n"
+            )
+            for module in unavailable_modules:
+                description += "- " + module + "\n"
+            theme_not_found = any(module[:6] == "theme_" for module in unavailable_modules)
```

Comment:
enterprise_not_found = any(module[:6] != "theme_" for module in unavailable_modules)

Reply:
I couldn't find something simpler and more efficient than
```python
            enterprise_not_found = any(
                module[:6] != "theme_" and not self.search([('name', '=', module)], limit=1)
                for module in unavailable_modules
            )
```
Or we stick to the assumption that an enterprise module (knowledge) will always be present in any industry module, and we can simply do as before
```python
            enterprise_not_found = release.version[-1] != 'e'
```